### ISSUE TITLE
Single sign-out asp.net

### ISSUE THREAD

**Issue by Astimus** [IdentityServer/IdentityServer3#3602]
I have two clients:
1) asp.net core
2) js angular application (oidc-token-manager)

In IdSrv3 configuration I've set LogoutUri for asp.net core application and added method to home controller SignoutCleanup as described in instructions. But when I logout in my Angular application IdSrv not add <iframe> with asp.net core application LogoutUri. Breakpoint inside SignoutCleanup method of HomeController not hitting. What I need to check? Nothing interesting in logs. I use SEQ logging.

**Comment by brockallen** [comment]
Not sure -- you'd need to really debug into the generated iframe and any child iframes in that one.

**Comment by Astimus** [comment]
@brockallen That happens if I sign-out inside my asp.net core application too. SignoutCleanup not invoking in my HomeController inside asp.net core application. How to debug this thing?

**Comment by brockallen** [comment]
Do you have a LogoutUri set for the MVC client (which is different than the PostLogotRedirectUri)?

**Comment by Astimus** [comment]
@brockallen There is no PostLogoutRedirectUri set for MVC client.
`ClientName = "Viewer",
                    Enabled = true,
                    ClientId = "viewer",
                    Flow = Flows.Implicit,
                    RedirectUris = new List<string> { $"{ViewerUri}/signin-oidc" },
                    RequireSignOutPrompt = false,
                    RequireConsent = false,
                    LogoutUri = $"{ViewerUri}/Home/SignoutCleanup",
                    LogoutSessionRequired = true,
                    AllowedScopes = new List<string>
                    {
                        Constants.StandardScopes.OpenId,
                        Constants.StandardScopes.Profile,
                        Constants.StandardScopes.Email,
                        Constants.StandardScopes.Roles,
                        Constants.StandardScopes.Address,
                        "core_facade_logic",
                        "script_service"
                    },
                    AccessTokenLifetime = 90000`

**Comment by brockallen** [comment]
Then not sure what the issue is. Do your IS logs say anything about the MVC client app during the user signout (such as `Client end session iframe URLs:...` or `No client end session iframe URLs`)?

**Comment by Astimus** [comment]
I have problem with signout in asp. core application. 
I call (MVC method adapted for asp.net core):
```
        public ActionResult Signout()
        {
            Request.GetOwinContext().Authentication.SignOut();
            return Redirect("/");
        }
```
And no logs in SEQ about signout.

**Comment by brockallen** [comment]
> Request.HttpContext.Authentication.SignOutAsync("Cookies");

Is this how our samples are doing it?

**Comment by Astimus** [comment]
@brockallen 
In documentation: 
```
public ActionResult Signout()
        {
            Request.GetOwinContext().Authentication.SignOut();
            return Redirect("/");
        }
```
But in asp.net core there is no GetOwinContext(). I use HttpContext instead.

**Comment by brockallen** [comment]
MVC Core sample client: https://github.com/IdentityServer/IdentityServer4.Samples/blob/release/Clients/src/MvcHybrid/Controllers/HomeController.cs#L80

**Comment by Astimus** [comment]
@brockallen 
Will it work correctly with IdSrv3?

**Comment by brockallen** [comment]
Yes, as it's all just OIDC on the network.

**Comment by Astimus** [comment]
@brockallen Ok. I will check this way tommorow and will inform you.
But these problem is not related to the fact, that when I signout in my angular application (oidc-token-manager) SignoutCleanup method is not hit.

**Comment by brockallen** [comment]
>  But these problem is not related to the fact, that when I signout in my angular application (oidc-token-manager) SignoutCleanup method is not hit.

Well, if you are not redirected to IdentityServer, then you won't.

**Comment by Astimus** [comment]
When I logout in my angular application IdSrv redirect me to his logout page. And there is no logoutUri for asp.core application in ModelJson.

**Comment by brockallen** [comment]
As I suggested, correlate the logs to the code.

**Comment by Astimus** [comment]
@brockallen Problem was in my custom loggedOut.html. I've accidentally deleted ng-repeat of iframes inside this page.

**Comment by Disciplemike** [comment]
Hey, I know this may be an old issue, but if you made the mistake I made, I hope this helps.
Make sure your SignoutCleanup is NOT AUTHORIZED!  It seems pretty obvious, in hindsight, but it took me hours to figure this out.  
I had my application adding a filter of Authorize attribute to everything.  Hence, I didn't notice it on the controller method.
I just added [AllowAnonymous] to the SignoutCleanup method and blamo... everything works!

I hope this saves someone some time.
